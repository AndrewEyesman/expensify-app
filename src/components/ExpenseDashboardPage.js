import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDashboardpage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export default ExpenseDashboardpage