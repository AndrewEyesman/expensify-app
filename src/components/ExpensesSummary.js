import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import selectExpensesTotal from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses'

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => (
    <div>
        <p>
            {
                expenseCount > 0 ? (
                    expenseCount === 1 ? 
                    (`Viewing 1 expense totalling ${numeral(expenseTotal / 100).format('$0,0.00')}`) : 
                    (`Viewing ${expenseCount} expenses totalling ${numeral(expenseTotal / 100).format('$0,0.00')}`)
                ) : 'Add an expense to get started!'
            }
        </p>
    </div>
)

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)

    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: selectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)