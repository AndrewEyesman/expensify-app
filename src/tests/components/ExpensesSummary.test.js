import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'
import selectExpensesTotal from '../../selectors/expenses-total'

test('should render ExpensesSummary with no expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={0} expenseTotal={0}/>)

    expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={expenses[0].amount}/>)

    expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={expenses.length} expenseTotal={selectExpensesTotal(expenses)}/>)

    expect(wrapper).toMatchSnapshot()
})