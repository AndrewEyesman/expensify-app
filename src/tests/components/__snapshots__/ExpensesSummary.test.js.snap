// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing 1 expense totalling $1.95
  </p>
</div>
`;

exports[`should render ExpensesSummary with multiple expenses 1`] = `
<div>
  <p>
    Viewing 3 expenses totalling $87.83
  </p>
</div>
`;

exports[`should render ExpensesSummary with no expenses 1`] = `
<div>
  <p>
    Add an expense to get started!
  </p>
</div>
`;
